tests/features/GLOBAL.flags

```diff
@@ -1 +1 @@
--Werror-implicit-function-declaration
+-Werror-implicit-function-declaration -Wno-unused-variable
```

Comment:
Related to https://github.com/aws/s2n-tls/issues/4187

My new S2N_LINUX_SENDFILE feature flag was failing with "unused variable", and so was S2N_KTLS_SUPPORTED. Probably best just to ignore unused variables, since they have nothing to do with feature support. 

But if we think that's too risky, I can just fix both feature tests to not have unused variables. However, I suspect we'll keep making this mistake.